src/django_components/templatetags/component_tags.py

```diff
@@ -57,34 +136,192 @@ def _component_dependencies(type: Literal["js", "css"]) -> SafeString:
+    are by default inserted into the `<head>` tag of the HTML. (See
+    [JS and CSS output locations](../../concepts/advanced/rendering_js_css/#js-and-css-output-locations))
 
-@register.simple_tag(name="component_css_dependencies")
-def component_css_dependencies() -> SafeString:
-    """Marks location where CSS link tags should be rendered."""
+    Note that there should be only one `{% component_css_dependencies %}` for the whole HTML document.
+    If you insert this tag multiple times, ALL CSS links will be duplicately inserted into ALL these places.
+    """
+    # Parse to check that the syntax is valid
+    _parse_tag(parser, token, tag_spec)
     return _component_dependencies("css")
 
 
-@register.simple_tag(name="component_js_dependencies")
-def component_js_dependencies() -> SafeString:
-    """Marks location where JS script tags should be rendered."""
+@register.tag(name="component_js_dependencies")
+@with_tag_spec(
+    TagSpec(
+        tag="component_js_dependencies",
+        end_tag=None,  # inline-only
+    )
+)
+def component_js_dependencies(parser: Parser, token: Token, tag_spec: TagSpec) -> TextNode:
+    """
+    Marks location where JS link tags should be rendered after the whole HTML has been generated.
+
+    Generally, this should be inserted at the end of the `<body>` tag of the HTML.
+
+    If the generated HTML does NOT contain any `{% component_js_dependencies %}` tags, JS scripts
+    are by default inserted at the end of the `<body>` tag of the HTML. (See
+    [JS and CSS output locations](../../concepts/advanced/rendering_js_css/#js-and-css-output-locations))
+
+    Note that there should be only one `{% component_js_dependencies %}` for the whole HTML document.
+    If you insert this tag multiple times, ALL JS scripts will be duplicately inserted into ALL these places.
+    """
+    # Parse to check that the syntax is valid
+    _parse_tag(parser, token, tag_spec)
     return _component_dependencies("js")
 
 
 @register.tag("slot")
-def slot(parser: Parser, token: Token) -> SlotNode:
-    tag = _parse_tag(
-        "slot",
-        parser,
-        token,
-        params=[SLOT_NAME_KWARG],
-        optional_params=[SLOT_NAME_KWARG],
-        flags=[SLOT_DEFAULT_KEYWORD, SLOT_REQUIRED_KEYWORD],
-        keywordonly_kwargs=True,
-        repeatable_kwargs=False,
+@with_tag_spec(
+    TagSpec(
+        tag="slot",
         end_tag="endslot",
+        positional_only_args=[],
+        pos_or_keyword_args=[SLOT_NAME_KWARG],
+        keywordonly_args=True,
+        repeatable_kwargs=False,
+        flags=[SLOT_DEFAULT_KEYWORD, SLOT_REQUIRED_KEYWORD],
     )
+)
+def slot(parser: Parser, token: Token, tag_spec: TagSpec) -> SlotNode:
+    """
+    Slot tag marks a place inside a component where content can be inserted
+    from outside.
+
+    Similar to [Vue's slots](https://vuejs.org/guide/components/slots.html) or React's `children`.
```

Comment:
Or the Web Component with the same name: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot - Maybe link to our own docs first, and have these references there instead?

Reply:
Whaaaat, I didn't know Web components had slots! Such a shame that the content of web components is not queriable with `document.querySelector`, otherwise I'd be already using it :/

Changed the order and added link to web components.